dev/run-tests.py

```diff
@@ -0,0 +1,417 @@
+
+    # map over the values and convert them to integers
+    return map(lambda x: int(x), version.split('.') + [update])
+
+
+def multi_starts_with(orig_str, *prefixes):
+    """Takes a string and an abritrary number of prefixes then checks the
+    original string for any of the possible prefixes passed in"""
+
+    for s in prefixes:
+        if orig_str.startswith(s):
+            return True
+    return False
+
+
+def determine_test_suite():
+    """This function current acts to determine if SQL tests need to be run in
+    addition to the core test suite *or* if _only_ SQL tests need to be run
+    as the git logs show that to be the only thing touched. In the future
+    this function will act more generically to help further segregate the
+    test suite runner (hence the function name).
+    @return a set of unique test names"""
+    test_suite = list()
+
+    if amplab_jenkins:
+        run_cmd(['git', 'fetch', 'origin', 'master:master']).wait()
+
+        raw_output = subprocess.check_output(['git', 'diff', '--name-only', 'master'])
+        # remove any empty strings
+        changed_files = [f for f in raw_output.split('\n') if f]
+
+        # find any sql files
+        sql_files = [f for f in changed_files
+                     if multi_starts_with(f, 
+                                          "sql/", 
+                                          "bin/spark-sql", 
+                                          "sbin/start-thriftserver.sh")]
+
+        non_sql_files = set(changed_files).difference(set(sql_files))
+
+        if non_sql_files:
+            test_suite.append("CORE")
+        if sql_files:
+            print "[info] Detected changes in SQL. Will run Hive test suite."
+            test_suite.append("SQL")
+            if not non_sql_files:
+                print "[info] Detected no changes except in SQL. Will only run SQL tests."
+        return set(test_suite)
+    else:
+        # we aren't in the Amplab environment so merely run all tests
+        test_suite.append("CORE")
+        test_suite.append("SQL")
+        return set(test_suite)
+
+
+def set_title_and_block(title, err_block):
+    os.environ["CURRENT_BLOCK"] = error_codes[err_block]
+    line_str = "".join(['='] * 72)
+
+    print
+    print line_str
+    print title
+    print line_str
+
+
+def run_apache_rat_checks():
+    set_title_and_block("Running Apache RAT checks", "BLOCK_RAT")
+    run_cmd(["./dev/check-license"])
+
+
+def run_scala_style_checks():
+    set_title_and_block("Running Scala style checks", "BLOCK_SCALA_STYLE")
+    run_cmd(["./dev/lint-scala"])
+
+
+def run_python_style_checks():
+    set_title_and_block("Running Python style checks", "BLOCK_PYTHON_STYLE")
+    run_cmd(["./dev/lint-python"])
+
+
+def exec_maven(mvn_args=[]):
+    """Will call Maven in the current directory with the list of mvn_args passed
+    in and returns the subprocess for any further processing"""
+
+    return subprocess.Popen(["./build/mvn"] + mvn_args)
```

Comment:
Is there any reason to not just `wait()` here? It seems like that's what we're always doing with the return value.

Also, if we want to raise an exception when this fails (which I think we do), we should change `Popen()` to `check_call()`, in which case no need for the `wait()` either.